data/update_script.sh.erb

```diff
@@ -51,16 +53,24 @@ echo 'PULLPREVIEW_PUBLIC_IP=<%= locals.public_ip %>' >> $PULLPREVIEW_ENV_FILE
 echo 'PULLPREVIEW_URL=<%= locals.url %>' >> $PULLPREVIEW_ENV_FILE
 echo "PULLPREVIEW_FIRST_RUN=$PULLPREVIEW_FIRST_RUN" >> $PULLPREVIEW_ENV_FILE
 
+if [ -n "${PULLPREVIEW_ENV_VARS}" ] ; then
+while read -d, -r pair; do
+  IFS='=' read -r key val <<<"$pair"
+  echo "$key=$val" >> $PULLPREVIEW_ENV_FILE
+done <<<"$PULLPREVIEW_ENV_VARS,"
+fi
+
 set -o allexport
 source $PULLPREVIEW_ENV_FILE
 set +o allexport
 
 cd /
 
-sudo rm -rf "$APP_PATH"
-sudo mkdir -p "$APP_PATH"
-sudo chown -R ec2-user.ec2-user "$APP_PATH"
-tar xzf "$1" -C "$APP_PATH"
+sudo rm -rf "/tmp$APP_PATH"
+sudo mkdir -p "$APP_PATH" "/tmp$APP_PATH"
+sudo chown -R ec2-user.ec2-user "$APP_PATH" "/tmp$APP_PATH"
+tar xzf "$1" -C "/tmp$APP_PATH"
+rsync -auz "/tmp$APP_PATH" / --remove-source-files --delete
```

Comment:
Why not rsync to the host in the first place? Anyway, this got rid of the situation that before the folder was replaced leading to weird side effects when being logged into the shell of the docker container or the app folder on the host